### ISSUE TITLE
VSProjectExternals not possible to put in solution folders

### ISSUE THREAD

**Issue by jespergustinmsft** [fastbuild/fastbuild#778]
Adding VSProjectExternal() projects to VSSolution -> SolutionFolders -> Projects array does not work as expected. Project ends up in solution root.

Example:
```
VSProjectExternal( 'Project1' )
{
    .ExternalProjectPath = 'Project1.vcxproj'
}

.FolderProjectExample = [
    .Path      = 'Example'
    .Projects = { 'Project1' }
]

VSSolution( 'ExampleSolution-sln' )
{
    .SolutionOutput = 'ExampleSolution.sln'
    .SolutionFolders = {
        .FolderProjectExample
    }
}
```

Example above causes Project1 to end up in solution root, but expected to be located under a subfolder called Example.

**Comment by ffulin** [comment]
At a glance, your bff setup seems correct, so I don't see an obvious reason for this not workin. I'll need to investigate a little to try and repro the issue.

**Comment by ffulin** [comment]
@jespergustinmsft I tried to reproduce this problem, but in my test, the project appears in the solution under an "Example" folder as expected.

I created a generic vcxproj using VS2019 as the "Project1.vcxproj" and used your example bff, but manually specified the .ProjectTypeGuid and .ProjectConfigs.  (I also tried using the 3rd party VSProjExtractor dll, but the most recent 2 version was crash for me for reasons unknown). I used FASTBuild v1.02.

I suspect there is some problem working out the relative directories, so it might depend on where your files live. If possible, are you able to help me find a repro by telling me:
 1) the FASTBuild version you are using
 2) anything interesting about the external project you were using
 3) what path you are running in
 4) whether the problem only occurs when using the VSProjTypeExtractor dll (which I assume you are using) or whether it still occurs if you specify the .ProjectTypeGuid and .ProjectConfigs as I did instead.

If possible, a zip file containing a stand-alone repro would be ideal as well.
